### ISSUE TITLE
Cannot convert from JSON to MongoDB

### ISSUE THREAD

**Issue by laggron42** [Cog-Creators/Red-DiscordBot#2251]
# Other bugs

#### What were you trying to do?

Edit an instance to modify its storage type from JSON to MongoDB.

#### What were you expecting to happen?

Convert the JSON data to the MongoDB scheme.

#### What actually happened?

```py
Please choose your storage backend (if you re unsure, choose 1).
1. JSON (file storage, requires no database).
2. MongoDB
> 2
Enter URI scheme (mongodb or mongodb+srv): mongodb
Enter host address: 127.0.0.1:27017
Enter login username:
Enter login password:
Enter mongodb database name: Oxibot
Would you like to import your data? (y/n) y
Traceback (most recent call last):
  File "/usr/local/bin/redbot-setup", line 11, in <module>
    load_entry_point('Red-DiscordBot==3.0.0rc2', 'console_scripts', 'redbot-setup')()
  File "/usr/local/lib/python3.6/dist-packages/redbot/setup.py", line 384, in main
    loop.run_until_complete(edit_instance())
  File "/usr/lib/python3.6/asyncio/base_events.py", line 467, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.6/dist-packages/redbot/setup.py", line 278, in edit_instance
    await json_to_mongo(current_data_dir, storage_details)
  File "/usr/local/lib/python3.6/dist-packages/redbot/setup.py", line 198, in json_to_mongo
    {"_id": cog_m.unique_cog_identifier}, update={"$set": cog_data[cog_i]}, upsert=True
KeyError: '-4851999857783813997'
```

#### How can we reproduce this issue?

According to Crossedfall, this seems to happen with 3rd party cogs, but I can't give more details, the error does not details which file the bot was editing, and there's no logger in `redbot-setup`.

**Comment by sleepymanakete** [comment]
I am getting an error similar to the one above doing the same steps.

```Traceback (most recent call last):
  File "/home/lailah/bots/bin/redbot-setup", line 11, in <module>
    load_entry_point('Red-DiscordBot==3.0.0rc2', 'console_scripts',
'redbot-setup')()
  File "/home/lailah/bots/lib/python3.6/site-packages/redbot/setup.py",
line 384, in main
    loop.run_until_complete(edit_instance())
  File "/usr/lib/python3.6/asyncio/base_events.py", line 468, in
run_until_complete
    return future.result()
  File "/home/lailah/bots/lib/python3.6/site-packages/redbot/setup.py",
line 278, in edit_instance
    await json_to_mongo(current_data_dir, storage_details)
  File "/home/lailah/bots/lib/python3.6/site-packages/redbot/setup.py",
line 198, in json_to_mongo
    {"_id": cog_m.unique_cog_identifier}, update={"$set":
cog_data[cog_i]}, upsert=True
KeyError: '-2039828293'```

**Comment by Kowlin** [comment]
Can confirm that this is an issue.

**Comment by Tobotimus** [comment]
Resolved by #2346.
